src/Client/ClientBase.cpp

```diff
@@ -1748,7 +1748,8 @@ void ClientBase::sendData(Block & sample, const ColumnsDescription & columns_des
     }
     else if (!is_interactive)
     {
-        sendDataFromStdin(sample, columns_description_for_query, parsed_query);
+        if (!is_local)
+            sendDataFromStdin(sample, columns_description_for_query, parsed_query);
```

Comment:
Could you explain why this helps to solve the issue?

Reply:
This logic handles the specific logic of the derived class in the base class.

I think we can create a virtual function to separate the logic of the base and derived class:

```
// in ClientBase
virtual void sendDataWithNonInteractiveInterface(...) {
    sendDataFromStdin(sample, columns_description_for_query, parsed_query);
}

// in LocalServer
void sendDataWithNonInteractiveInterface(...) override {
    // Do nothing
}
```

In `else if (!is_interactive)` block, we can just call `sendDataWithNonInteractiveInterface(...)`. And we don't need to introduce `is_local` in `ClientBase`.